VirtualEconomyFramework/VEFrameworkUnitTest/Neblio/SendNeblioTransactionTest.cs

```diff
@@ -245,11 +251,146 @@ public void SendNeblioTransaction_CannotSignTransaction_Error_Test()
+
+            SendTxData sendTxData = new SendTxData()
+            {
+                ReceiverAddress = address,
+                SenderAddress = address,
+                Amount = 1, 
+                CustomMessage = "test",
+                Password = ""
+            };
+
+            foreach(var u in addressObject.Utxos)
+                 u.Index = 2;
+
+            var transaction = await NeblioTransactionHelpers.SendNeblioTransactionAPIAsync(sendTxData, AccountKey, addressObject.Utxos);
+
+            Assert.Equal(10, transaction.Inputs.Count);
+            Assert.Equal(2, (int)transaction.Inputs[0].PrevOut.N);
+            Assert.Equal(NBitcoin.uint256.Parse(addressObject.Utxos.FirstOrDefault().Txid), 
+                                                transaction.Inputs[0].PrevOut.Hash);
+        }
+
+        /// <summary>
+        /// Unit test method to verify if outputs are correct
+        /// </summary>
+        [Fact]
+        public async void SendNeblioTransaction_CorrectNumberOfOutputsInTx_Test()
+        {
+            //Arrange           
+
+            var res = Common.FakeDataGenerator.GetKeyAndAddress();
+
+            string address = res.Item1.ToString();
+            string key = res.Item2.ToString();
+            var AccountKey = new EncryptionKey(key);
+
+            GetAddressInfoResponse addressObject = Common.FakeDataGenerator.GetAddressWithNeblUtxos(address, 10, 100000000);
+
+            SendTxData sendTxData = new SendTxData()
+            {
+                ReceiverAddress = address,
+                SenderAddress = address,
+                Amount = 1,
+                CustomMessage = "",
+                Password = ""
+            };
+
+            var expectedFee = 0.0002;
+            var totInputs = 0.0;
+            foreach (var utxo in addressObject.Utxos)
+                totInputs += utxo.Value.Value;
+
+            totInputs /= NeblioTransactionHelpers.FromSatToMainRatio;
+
+            var transaction = await NeblioTransactionHelpers.SendNeblioTransactionAPIAsync(sendTxData, AccountKey, addressObject.Utxos);
+
+            // we can expect use of two outputs for this tx. One to cover amount. second for rest of money back to source
+            Assert.Equal(2, transaction.Outputs.Count);
+            // Amount in the 0 output must be same as requested amount to send
+            Assert.Equal(sendTxData.Amount, Convert.ToDouble(transaction.Outputs[0].Value.ToUnit(NBitcoin.MoneyUnit.BTC)));
+            // 8.9998 is the rest after send 1 and fee 0.0002 in this tx - lots of inputs
+            Assert.Equal(totInputs - sendTxData.Amount - expectedFee, 
+                         Convert.ToDouble(transaction.Outputs[1].Value.ToUnit(NBitcoin.MoneyUnit.BTC)));
+        }
+
+
+        /// <summary>
+        /// Unit test method to verify if system is adding message to tx correctly
+        /// </summary>
+        [Fact]
+        public async void SendNeblioTransaction_MessageInNeblioTx_Test()
```

Comment:
I have created some new tests. Please try to create some similar test which test Neblio transaction. 

Please skip now the token transactions. Most of them uses the Neblio API now, to get basic structure.